### PR TITLE
[Resources and support] Implement "fieldAudienceBeneficiares" into search results and article listings

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
This PR begins generating article listings pages using the `fieldAudienceBeneficiares` property. It also implements that field into the search results React app.

https://github.com/department-of-veterans-affairs/va.gov-team/issues/14288
https://github.com/department-of-veterans-affairs/va.gov-team/issues/14841

## Testing done
Used the search results locally and confirmed the listings are being generated

## Screenshots
image

## Acceptance criteria
- [ ] `fieldAudienceBeneficiares` is implemented into search results
- [ ] Article listings are generated for `fieldAudienceBeneficiares` property

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/resources-and-support/components/SearchResultList.jsx [hunk 1]
```diff
@@ -16,51 +16,58 @@ const articleTypes = {
...
-                    (fieldTopic, index) => {
-                      return (
-                        <Tag key={`topic-${index}`} term={fieldTopic.entity} />
-                      );
-                    },
-                  )}
-                </div>
-              )}
-            </div>
+            <SearchResult article={article} />
```
Comment (ncksllvn) [inline src/applications/resources-and-support/components/SearchResultList.jsx:70]: The conditional rendering of the different fields was getting kinda wild so I separated out the search result into its own component.

File: src/site/stages/build/plugins/create-resources-and-support-section.js [hunk 2]
```diff
@@ -46,11 +46,17 @@ function groupByTags(allArticles) {
...
     } = article;
 
-    fieldTopics.map(fieldTopic => fieldTopic.entity).forEach(fieldTopic => {
+    const terms = [...fieldTopics];
+
+    if (fieldAudienceBeneficiares) {
+      terms.push(fieldAudienceBeneficiares);
+    }
+
+    terms.map(fieldTopic => fieldTopic.entity).forEach(fieldTopic => {
```
Comment (ncksllvn) [inline src/site/stages/build/plugins/create-resources-and-support-section.js:59]: Here's where `fieldAudienceBeneficiares` is processed into article listings pages like `http://localhost:3001/taxonomy/term/268/`

File: src/applications/resources-and-support/components/SearchResultList.jsx [hunk 3]
```diff
@@ -16,51 +16,58 @@ const articleTypes = {
   [ENTITY_BUNDLES.STEP_BY_STEP]: 'Step by step',
 };
 
+function SearchResult({ article }) {
```
Comment (kelsonic) [inline src/applications/resources-and-support/components/SearchResultList.jsx:19]: Should we move this into its own file you think? Up to you 🙂

Comment (ncksllvn) [reply src/applications/resources-and-support/components/SearchResultList.jsx:19]: Yeah I suppose we should

**Review by kelsonic — APPROVED** [review]
LGTM! 🎉

**Review by rianfowler — APPROVED** [review]

**Pull request merged**